### ISSUE TITLE
Secret var

### ISSUE THREAD

**Issue by MarceloBuzzettiToolbox** [google-github-actions/deploy-cloudrun#528]
### TL;DR

I try to set docker registry, docker.io, secret and use to deploy but not worked.
 image: "${{ secrets.DOCKER_REGISTRY }}/${{ secrets.DOCKER_NAMESPACE }}/${{ secrets.DOCKER_REPOSITORY }}:latest"

When i created it as repository variable worked
          image: "${{ vars.DOCKER_REGISTRY }}/${{ secrets.DOCKER_NAMESPACE }}/${{ secrets.DOCKER_REPOSITORY }}:latest"

Before asking me, I created the secret and repository variable DOCKER_REGISTRY correctly.

### Expected behavior

Run using registry as secret

### Observed behavior

Just work when used as repository variable

### Action YAML

```yaml
- name: Deploy on Cloud Run # Faz o deploy da aplicação no Cloud Run
        uses: 'google-github-actions/auth@v2' # Usa a ação de autenticação do Google Cloud
        with: # Define os parâmetros da ação
          workload_identity_provider: ${{ secrets.WORKLOAD_IDENTIFIER_PROVIDER }} # Define o provedor de identidade do workload
          service_account: ${{ secrets.SERVICE_ACCOUNT }} # Define a conta de serviço

      - id: 'deploy' # Define um ID para o passo
        uses: 'google-github-actions/deploy-cloudrun@v2' # Usa a ação de deploy no Cloud Run
        with: # Define os parâmetros da ação
          service: 'nodejs-toolbox-playground-exemplo' # Define o nome do serviço no Cloud Run
          image: "${{ secrets.DOCKER_REGISTRY }}/${{ secrets.DOCKER_NAMESPACE }}/${{ secrets.DOCKER_REPOSITORY }}:latest" # Define a imagem a ser usada no Cloud Run
```

### Log output

```text
google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy nodejs-toolbox-playground-exemplo --image ***/***/***:latest --update-labels ^,^managed-by=github-actions,commit-sha=6d33c3b6c54a2c6e27631cba0b51ab00960922bf --format json --region us-central1`: Deploying container to Cloud Run service [nodejs-toolbox-playground-exemplo] in project [toolbox-sandbox-388523] region [us-central1]
```

**Comment by sethvargo** [comment]
Hi there - could you please provide the [debug output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) for the complete GitHub Actions workflow run?

**Comment by MarceloBuzzettiToolbox** [comment]
> Hi there - could you please provide the [debug output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) for the complete GitHub Actions workflow run?

﻿2024-06-11T21:40:22.2227195Z ##[group]Run google-github-actions/deploy-cloudrun@v2
with:
  service: nodejs-toolbox-playground-exemplo
  image: ***/***/***:latest
  env_vars_update_strategy: merge
  secrets_update_strategy: merge
  skip_default_labels: false
  no_traffic: false
  region: us-central1
env:
  DOCKER_METADATA_OUTPUT_VERSION: latest
  DOCKER_METADATA_OUTPUT_TAGS: ***/***:latest
***/***:31
  DOCKER_METADATA_OUTPUT_LABELS: org.opencontainers.image.created=2024-06-11T21:40:02.452Z
org.opencontainers.image.description=Exemplos de configuração de pipelines
org.opencontainers.image.licenses=NOASSERTION
org.opencontainers.image.revision=6d33c3b6c54a2c6e27631cba0b51ab00960922bf
org.opencontainers.image.source=https://github.com/toolbox-playground/pipelines-exemplo-basico
org.opencontainers.image.title=pipelines-exemplo-basico
org.opencontainers.image.url=https://github.com/toolbox-playground/pipelines-exemplo-basico
org.opencontainers.image.version=latest
  DOCKER_METADATA_OUTPUT_ANNOTATIONS: manifest:org.opencontainers.image.created=2024-06-11T21:40:02.452Z
manifest:org.opencontainers.image.description=Exemplos de configuração de pipelines
manifest:org.opencontainers.image.licenses=NOASSERTION
manifest:org.opencontainers.image.revision=6d33c3b6c54a2c6e27631cba0b51ab00960922bf
manifest:org.opencontainers.image.source=https://github.com/toolbox-playground/pipelines-exemplo-basico
manifest:org.opencontainers.image.title=pipelines-exemplo-basico
manifest:org.opencontainers.image.url=https://github.com/toolbox-playground/pipelines-exemplo-basico
manifest:org.opencontainers.image.version=latest
  DOCKER_METADATA_OUTPUT_JSON: {"tags":["***/***:latest","***/***:31"],"labels":{"org.opencontainers.image.created":"2024-06-11T21:40:02.452Z","org.opencontainers.image.description":"Exemplos de configuração de pipelines","org.opencontainers.image.licenses":"NOASSERTION","org.opencontainers.image.revision":"6d33c3b6c54a2c6e27631cba0b51ab00960922bf","org.opencontainers.image.source":"https://github.com/toolbox-playground/pipelines-exemplo-basico","org.opencontainers.image.title":"pipelines-exemplo-basico","org.opencontainers.image.url":"https://github.com/toolbox-playground/pipelines-exemplo-basico","org.opencontainers.image.version":"latest"},"annotations":["manifest:org.opencontainers.image.created=2024-06-11T21:40:02.452Z","manifest:org.opencontainers.image.description=Exemplos de configuração de pipelines","manifest:org.opencontainers.image.licenses=NOASSERTION","manifest:org.opencontainers.image.revision=6d33c3b6c54a2c6e27631cba0b51ab00960922bf","manifest:org.opencontainers.image.source=https://github.com/toolbox-playground/pipelines-exemplo-basico","manifest:org.opencontainers.image.title=pipelines-exemplo-basico","manifest:org.opencontainers.image.url=https://github.com/toolbox-playground/pipelines-exemplo-basico","manifest:org.opencontainers.image.version=latest"]}
  DOCKER_METADATA_OUTPUT_BAKE_FILE_TAGS: /home/runner/work/_temp/docker-actions-toolkit-mq8vO3/docker-metadata-action-bake-tags.json
  DOCKER_METADATA_OUTPUT_BAKE_FILE_LABELS: /home/runner/work/_temp/docker-actions-toolkit-mq8vO3/docker-metadata-action-bake-labels.json
  DOCKER_METADATA_OUTPUT_BAKE_FILE_ANNOTATIONS: /home/runner/work/_temp/docker-actions-toolkit-mq8vO3/docker-metadata-action-bake-annotations.json
  DOCKER_METADATA_OUTPUT_BAKE_FILE: /home/runner/work/_temp/docker-actions-toolkit-mq8vO3/docker-metadata-action-bake.json
  CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/pipelines-exemplo-basico/pipelines-exemplo-basico/gha-creds-960582e71cfd3a2d.json
  GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/pipelines-exemplo-basico/pipelines-exemplo-basico/gha-creds-960582e71cfd3a2d.json
  GOOGLE_GHA_CREDS_PATH: /home/runner/work/pipelines-exemplo-basico/pipelines-exemplo-basico/gha-creds-960582e71cfd3a2d.json
  CLOUDSDK_CORE_PROJECT: toolbox-sandbox-388523
  CLOUDSDK_PROJECT: toolbox-sandbox-388523
  GCLOUD_PROJECT: toolbox-sandbox-388523
  GCP_PROJECT: toolbox-sandbox-388523
  GOOGLE_CLOUD_PROJECT: toolbox-sandbox-388523
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/aa4f10cc-71a0-4189-8d0d-072700a32e4f -f /home/runner/work/_temp/8fd055ba-cd29-45df-9b11-d34f5d745494
Successfully authenticated
Running: gcloud run deploy nodejs-toolbox-playground-exemplo --image ***/***/***:latest --update-labels ^,^managed-by=github-actions,commit-sha=6d33c3b6c54a2c6e27631cba0b51ab00960922bf --format json --region us-central1
google-github-actions/deploy-cloudrun failed with: failed to execute gcloud command `gcloud run deploy nodejs-toolbox-playground-exemplo --image ***/***/***:latest --update-labels ^,^managed-by=github-actions,commit-sha=6d33c3b6c54a2c6e27631cba0b51ab00960922bf --format json --region us-central1`: Deploying container to Cloud Run service [nodejs-toolbox-playground-exemplo] in project [toolbox-sandbox-388523] region [us-central1]
Deploying...
failed
Deployment failed
ERROR: (gcloud.run.deploy) spec.template.spec.containers[0].image: Expected an image path like [host/]repo-path[:tag and/or @digest], where host is one of [region.]gcr.io, [region-]docker.pkg.dev or docker.io but obtained ***/***/***:latest

**Comment by MarceloBuzzettiToolbox** [comment]
I'll enable that and send to you

**Comment by MarceloBuzzettiToolbox** [comment]
> Hi there - could you please provide the [debug output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging) for the complete GitHub Actions workflow run?

https://productionresultssa1.blob.core.windows.net/actions-results/ea711417-9c77-4049-b46a-b6d444c2ca93/workflow-job-run-efc0764b-17ce-5a5a-2aed-65d2435877c6/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-06-11T23%3A40%3A08Z&sig=yMHqGznB5cK1eWg3JIXxbRFX6w0z3d6XiSzggjzPrng%3D&ske=2024-06-12T10%3A45%3A54Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-06-11T22%3A45%3A54Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2023-11-03&sp=r&spr=https&sr=b&st=2024-06-11T23%3A30%3A03Z&sv=2023-11-03

**Comment by MarceloBuzzettiToolbox** [comment]
Em ter, 11 de jun de 2024 20:03, Seth Vargo ***@***.***>
escreveu:

**Comment by MarceloBuzzettiToolbox** [comment]
[logs_24792883202.zip](https://github.com/user-attachments/files/15796249/logs_24792883202.zip)

**Comment by sethvargo** [comment]
What are the values for `DOCKER_REGISTRY`, `DOCKER_NAMESPACE`, and `DOCKER_REPOSITORY`? In general, these are not "secrets", so you might want to use `vars` instead so they aren't masked in the output. The value should be something like:

```
us-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner
```

**Comment by MarceloBuzzettiToolbox** [comment]
> What are the values for `DOCKER_REGISTRY`, `DOCKER_NAMESPACE`, and `DOCKER_REPOSITORY`? In general, these are not "secrets", so you might want to use `vars` instead so they aren't masked in the output. The value should be something like:
> 
> ```
> us-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner
> ```

These are values about the repository on docker hub. I thought it was necessary to keep them secret. Thanks for the feedback. I'll use them as vars instead of secret
